Project cleanup

wo80 opened the pull request:
This pull request does some cleanup of the project files:

* Remove legacy .NET Framework 4.x targets from main project and add net6.0 target
* Remove unnecessary package references
* Update package references
* Remove AssemblyInfo.cs from main project
* Fix the unit test project, disabling one failing test (Github issue 14)

I marked this as draft, because before merging

1. I'd like to remove AssemblyInfo.cs from the test project, but there's an irritating copyright notice pointing to Hewlett-Packard. What's that about? Can it be removed?
2. I'd like to remove all stuff related to Nuget config. Packaging should be accessible from the dotnet command line, so no need to have a Nuget executable in the repository.
3. I'd like to change the folder structure, so that instead of having the projects in `Source/Griffin.WebServer/` we'd just have a `src/` folder.
4. I'd like to add a Github workflow, thus removing the Appveyor dependency

@jgauffin Please let me know which of the changes you approve and whether they should go into this pull request or a separate.

wo80 committed:
Update gitignore.

When cloning a repository with Visual Studio, it will create a .vs folder in the top level directory. This change will exclude all those folders, regardless of the location.

wo80 committed:
Remove legacy .NET Framework 4.x targets.

wo80 committed:
Remove unnecessary package references.

wo80 committed:
Update nuget packages.

wo80 committed:
Remove AssemblyInfo.cs and use project file to specify package data.

wo80 committed:
Remove app.config from DemoServer project.

wo80 committed:
Fix unit test project.

wo80 commented:
Btw, locally the test are passing. So there seems to be some problem with the Appveyor setup.

jgauffin commented:
Yep, I hadn't updated visual studio version. Have done that now and started a new build.

wo80 committed:
Remove AssemblyInfo.cs from demo and test projects.

wo80 committed:
Delete .nuget folder and remove Nuget config from project files.

wo80 committed:
Delete Nuget package.

Makes no sense to have it in the source tree. Either make a release or keep the package on Nuget up-to-date.

wo80 commented:
I addressed point 1) and 2) from the first post. If you want me to revert some of this, let me know.

The Appveyor build still fails:
> Input file does not exist: src\Griffin.Framework\Griffin.Framework.sln

Obviously, `Griffin.Framework.sln` is nowhere in this repo, so still the problem must be the Appveyor setup.

wo80 committed:
Update Griffin.WebServer.csproj
* Add license identifier (see #45)
* Add package tags
* Create documentation file in output dir
